apps/payments/main.py

```diff
@@ -138,6 +182,18 @@ def payment_refund_request(payment_id, currency=None):
             )
             return redirect(url_for("users.purchases"))
 
+    for f in form.purchases:
+        update_refund_purchase_form_details(f, purchases_dict[f.purchase_id.data])
+
+    if get_refund_state() == "cancellation":
+        return render_template(
+            "payments/event-cancelled-refund-request.html",
+            payment=payment,
+            form=form,
+            currency=currency,
+            bank_validation_failed=bank_validation_failed,
+        )
+
     return render_template(
         "payments/refund-request.html",
```

Comment:
It'd be nice to retain a single template for these two if that's possible.  Understood if it's becoming too difficult to manage the variations using conditions, though.

Reply:
I may revisit this but I'd like to leave it as is for the time being